R/reportAgg.R

```diff
@@ -282,3 +301,56 @@ reportAgg <- function(x,
   }
   return(tags)
 }
+
+
+
+#' Split dimension names
+#'
+#' @param x character vector
+#' @param split character used to split \code{x}
+#' @returns character vector with each dimension as an own entry
+
+.split <- function(x, split = "\\.") {
+  if (is.null(x)) {
+    return(NULL)
+  }
+  unlist(strsplit(x, split))
+}
+
+
+
+#' select values from MAgPIE-obect
+#'
+#' This is a wrapper around \code{magclass::mselect} that also allows the
+#' selection of combinations of multiple dimensions.
+#'
+#' @param x MAgPIE object
+#' @param ... entry selection. Combined dimensions have to be seperated with
+#'   \code{.} for both the set names and the set elements.
+#' @returns MAgPIE object containing only selected entries
+
+.select <- function(x, ...) {
+  dims <- as.list(...)
+  combinedDims <- grep("\\.", names(dims), value = TRUE)
+  singleDims <- setdiff(names(dims), combinedDims)
+
+  if (length(singleDims) > 0) {
+    x <- mselect(x, dims[singleDims])
+  }
+
+  if (length(combinedDims) > 0) {
+    x <- Reduce(x = combinedDims, init = x, f = function(data, cd) {
+      patternRaw <- rep("[a-zA-Z0-9-]+", ndim(data, 3))
+      names(patternRaw) <- tail(getSets(data), -2)
+      regex <- paste(unlist(lapply(dims[[cd]], function(elem) {
+        pattern <- patternRaw
+        pattern[.split(cd)] <- str_escape(.split(elem))
+        paste(pattern, collapse = "\\.")
+      })), collapse = "|")
+      items <- grep(regex, getItems(x, 3), value = TRUE)
+      data[, , items]
+    })
+  }
```

Comment:
As discussed, these lines might also profit from additional comments, e.g. explaining the purpose of ```patternRaw```.

Reply:
I added in-line documentation and hope this makes it clearer.